File: src/Cxx/Math/MatrixTranspose.cxx
#include <vtkSmartPointer.h>
#include <vtkMatrix3x3.h>
	
int main(int, char *[])
{
  vtkSmartPointer<vtkMatrix3x3> m = 
    vtkSmartPointer<vtkMatrix3x3>::New();
  
  m->SetElement(2,1,2.0); // Set element (2,1) to 2.0
  
  std::cout << *m << std::endl;
  
  m->Transpose();
  
  std::cout << *m << std::endl;
  
  return EXIT_SUCCESS;
}

File: src/Java/Math/MatrixTranspose.java
import vtk.vtkNativeLibrary;
import vtk.vtkMatrix3x3;

public class MatrixTranspose 
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------

  public static void main(String args[]) 
  {
    // For a 3x3 matrices
    vtkMatrix3x3 Matrix = new vtkMatrix3x3();
    
    Matrix.SetElement(2, 1, 2.0); // Set element (0,0) to 1.0
    
    System.out.println(Matrix);
    
    Matrix.Transpose();
    
    System.out.println(Matrix); 
  }
}
